exchange/index.go

```diff
@@ -453,6 +476,145 @@ func (idx *Index) evict(size uint64) uint64 {
 	return evicted
 }
 
+func (idx *Index) tagForEviction(ref *DataRef) error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	return idx.walkDAG(ref.PayloadCID, func(blk blocks.Block) error {
+		idx.evictionSet.Add(blk.Cid())
+		return nil
+	})
+}
+
+func (idx *Index) GC() {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	// exit if there is nothing to evict
+	if idx.evictionSet.Len() == 0 {
+		return
+	}
+
+	gcbl, ok := idx.bstore.(blockstore.GCBlockstore)
+	if ok {
+		unlock := gcbl.GCLock()
+		defer unlock.Unlock()
+	}
+
+	err := idx.evictionSet.ForEach(func(c cid.Cid) error {
+		return idx.bstore.DeleteBlock(c)
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("failed to run garbage collector")
+	}
+
+	idx.evictionSet = cid.NewSet()
+}
+
+func (idx *Index) CleanBlockStore() error {
```

Comment:
```suggestion
func (idx *Index) CleanBlockStore(ctx context.Context) error {
```
Need a context to cleanup and close the channel etc